src/scheduler.c

```diff
@@ -84,15 +84,6 @@ JL_DLLEXPORT int jl_set_task_threadpoolid(jl_task_t *task, int8_t tpid) JL_NOTSA
 extern int jl_gc_mark_queue_obj_explicit(jl_gc_mark_cache_t *gc_cache,
                                          jl_gc_markqueue_t *mq, jl_value_t *obj) JL_NOTSAFEPOINT;
 
-// parallel task runtime
-// ---
-
-JL_DLLEXPORT uint32_t jl_rand_ptls(uint32_t max) // [0, n)
-{
-    jl_ptls_t ptls = jl_current_task->ptls;
-    return cong(max, &ptls->rngseed);
```

Comment:
```
src/threading.c
344:    return jl_current_task->ptls->rngseed;
350:    jl_current_task->ptls->rngseed = new_seed;
366:    ptls->rngseed = jl_rand();

src/gc-stock.h
449:    int v = gc_first_tid + (int)cong(jl_n_markthreads, &ptls->rngseed); // cong is [0, n)
```

So `ptls->rngseed` is still directly used by the GC.

Reply:
Right. But is that an issue?

Reply:
It's more that we have two slightly different algorithms use the same state variable, I was looking at where `cong` was being used

Reply:
They use the same rng update if that's any concern